src/libraries/Microsoft.PowerFx.Core/Binding/BindingConfig.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.PowerFx.Core.Binding
+{
+    /// <summary>
+    /// Configuration for an invocation of the binder.
+    /// </summary>
+    [ThreadSafeImmutable]
+    internal class BindingConfig
+    {
+        public static readonly BindingConfig Default = new BindingConfig(allowsSideEffects: false);
+
+        public bool AllowsSideEffects { get; }
+
+        public BindingConfig(bool allowsSideEffects)
+        {
+            AllowsSideEffects = allowsSideEffects;
```

Comment:
Is ConstantData not necessary either here?

Reply:
It is not, same logic as navigation. Those are going to live in PA Client's name resolver and be cast if needed.